include/revng/RestructureCFG/MarkerCallUtils.h

```diff
@@ -0,0 +1,222 @@
+    } else {
+      llvm::LLVMContext &C = getContext(BB);
+      llvm::PointerType *BlockAddressTy = llvm::Type::getInt8PtrTy(C);
+      auto
+        *NullBasicBlockAddress = llvm::ConstantPointerNull::get(BlockAddressTy);
+      Builder.CreateCall(MarkerCallF, NullBasicBlockAddress);
+    }
+  }
+};
+
+/// Helper method to retrieve the `BasicBlock` target of the marker
+inline llvm::BasicBlock *getMarkerCallTarget(std::string MarkerCallName,
+                                             llvm::BasicBlock *BB) {
+
+  // We must be provided with a `BasicBlock` where to search for the marker
+  revng_assert(BB);
+
+  // When using the getter, we assume that the marker function declaration is
+  // present in the `Module`
+  llvm::Module *M = getModule(BB);
+  llvm::Function *MarkerCallFunction = M->getFunction(MarkerCallName);
+  revng_assert(MarkerCallFunction);
+
+  // We assume that the call to the marker function can be in the last but one,
+  // or last but two position in the `BasicBlock`, without assuming a particular
+  // order between the marker call functions having a different type
+  auto BBIt = BB->rbegin();
+  ++BBIt;
+  for (size_t Index = 0; Index < 2 && BBIt != BB->rend(); ++BBIt) {
+    llvm::Instruction &TentativeInst = *BBIt;
+    if (llvm::CallInst *MarkerCall = getCallTo(&TentativeInst,
+                                               MarkerCallFunction)) {
+      auto *MarkerCallTargetBlockAddress = llvm::cast<
+        llvm::BlockAddress>(MarkerCall->getArgOperand(0));
+      using llvm::BasicBlock;
+      auto *MarkerCallTargetBB = MarkerCallTargetBlockAddress->getBasicBlock();
+      return MarkerCallTargetBB;
+    }
+  }
+
+  return nullptr;
+}
+
+inline llvm::BasicBlock *getScopeCloser(llvm::BasicBlock *BB) {
+  return getMarkerCallTarget(ScopeCloserMarkerCallFunctionName, BB);
+}
+
+inline bool isGotoBlock(llvm::BasicBlock *BB) {
+
+  // We must be provided with a `BasicBlock` where to search for the marker
+  revng_assert(BB);
+
+  // When using the getter, we assume that the marker function declaration is
+  // present in the `Module`
+  llvm::Module *M = getModule(BB);
+  llvm::Function
+    *MarkerCallFunction = getMarkerCallFunction(FunctionTags::GotoBlockMarker,
+                                                M);
+  revng_assert(MarkerCallFunction);
+
+  // We assume that the call to the marker function can be in the last but one,
+  // or last but two position in the `BasicBlock`, without assuming a particular
+  // order between the marker call functions having a different type
+  auto BBIt = BB->rbegin();
+  ++BBIt;
+  for (size_t Index = 0; Index < 2 && BBIt != BB->rend(); ++BBIt) {
+    llvm::Instruction &TentativeInst = *BBIt;
+    if (llvm::CallInst *MarkerCall = getCallTo(&TentativeInst,
+                                               MarkerCallFunction)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+inline void verifyBasicBlock(FunctionTags::Tag &Tag, llvm::BasicBlock *BB) {
```

Comment:
In order:
1) renaming has been done https://github.com/revng/revng/blob/57e9b483c79b868ba12fde41644a3ee0aba57651/include/revng/RestructureCFG/ScopeGraphUtils.h#L187
2) The main `verifyScopeGraphAnnotations` method, called by the `GraphTrait`, has no argument anymore. The `Impl` method called inside it, passes down the `Tag` just so in this way we can factor the common code https://github.com/revng/revng/blob/57e9b483c79b868ba12fde41644a3ee0aba57651/lib/RestructureCFG/ScopeGraphUtils.cpp#L14\
3) I have put this in a new `.cpp`, which gave origin to a new lib https://github.com/revng/revng/blob/57e9b483c79b868ba12fde41644a3ee0aba57651/lib/RestructureCFG/CMakeLists.txt#L6, since it needs to be linked also by the unit tests